### ISSUE TITLE
using behind corporate proxy

### ISSUE THREAD

**Issue by John-jk** [gmta/velocita-proxy#1]
Hello. 
Is it possible to use this library behind a corporate proxy server (using NAT is not possible for now)? At the moment I can't access packages on the Internet, because as far as I understand nginx is not able to explicitly use proxy servers for proxying.

As a result, the following scheme should be obtained: 
server with the application -> velocita-proxy  -> corporate proxy server (squid) -> repo.packagist.org | api.github.com | codeload.github.com ...

**Comment by gmta** [comment]
You're right that it currently is not possible with the configuration that velocita-proxy offers. I think that how it needs to work, is as follows:

* Configure an nginx upstream to the squid server
* Use `proxy_pass http://@squid-upstream` to redirect requests to the corp proxy
* Set the `Request-URI` HTTP header to the _target_ URI (not the Velocita URI) so Squid understands what URI to proxy to

Does that sound about right?

**Comment by John-jk** [comment]
I'm not sure, but I found a couple of articles on Google that also advised using nginx upstream , but I haven't been able to set it up that way yet. If you can confirm that it will work, I'll try it further.
